components/AppNav.tsx

```diff
@@ -69,9 +71,7 @@ const AuthButton: React.FC<AuthButtonProps> = ({ initial, username }) => {
   const [logoutUser, { data }] = useLogoutMutation()
   useEffect(() => {
     const success = _.get(data, 'logout.success', false)
-    if (success) {
-      window.location.pathname = '/'
-    }
+    if (success) window.location.pathname = '/'
```

Comment:
This originally used window.location, but can you change it to use `router.push` instead? The consistency in using the router object will make the codebase more clear

Reply:
It won't work. Without full refresh the page stays the same after logout. At first I thought it was because next uses fast refresh in development but production build is the same. 
I am not sure what options I need to pass to `router` to force full refresh.
The same thing with `login` page redirection.

Reply:
Oo thanks for detailed explanation! Things seem a lot more complicated than it appears to be on the surface. I'll try having a go at this and see if I can provide a solution

Reply:
I couldn't come up with a solution -- I guess nextjs `router.push` function doesn't like it when we pass in `/` as a value :(
No other simple solution came to mind as well, so it seems we are forced to use `window.location here`